src/RecursiveIteratorAggregateIterator.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace loophp\iterators;
+
+use Generator;
+use Iterator;
+use IteratorAggregate;
+use RuntimeException;
+use Traversable;
+
+use function count;
+
+/**
+ * @template TKey
+ * @template T
+ *
+ * @implements IteratorAggregate<int, T>
+ */
+class RecursiveIteratorAggregateIterator implements IteratorAggregate
+{
+    /**
+     * @var list<Iterator>
+     */
+    private ?array $stack = null;
+
+    /**
+     * @param Traversable<TKey,T> $input
+     */
+    public function __construct(private readonly Traversable $input) {}
+
+    public function getDepth(): int
```

Comment:
The `getDepth` method is not part of the `IteratorAggregate` interface, I would remove it.

Reply:
[getDepth](https://www.php.net/manual/en/recursiveiteratoriterator.getdepth) method is required for tree travel

Reply:
Then it should be private.

Reply:
It`s not used by iterator, its required to detect depth (or move direction) in loop. Like
```
foreach($it as $val){
   echo str_repeat('-', $it->getDepth()).$val."\n";
}
```

Reply:
yes I noticed it after your message. Since that method is not part of the interface, I would make it private.

Reply:
Or more common use case for xml/html/whatever
```
$prev_depth = 0;
foreach($it as $val){
   $next_depth = $it->getDepth();
   if ($prev_depth < $next_depth) echo "<tag>\n";
   for($i=$prev_depth;$i>$next_depth;$i--) echo "</tag>\n";
   echo $val;
   $prev_depth = $next_depth;
}
for($i=$prev_depth;$i>0;$i--) echo "</tag>\n";

```

Reply:
private method cannot be called outside of the iterator class (including foreach loop). It`s simple useless